### DISCUSSION TITLE
Adding more ways to stop from discarding unsaved work.

### DISCUSSION THREAD

**Discussion by KTibow** [home-assistant/frontend#7479] [Automations / scripts / scenes]
You probably don't want to discard your unsaved work. A nice little feature would be to register onbeforeunload, and show a popup when the window tries to be closed. Also, show a popup when you try to switch panels in the sidebar (but that's extra).
(In case you don't see the category, this is for automations/scripts/scenes)

**Comment by Recte** [comment]
It's there for a while now, but I'd like to address that the feature doesn't work well on IOS using the browser.
It shows 'stay' and 'leave' at the bottom but both just sent you back to the automation and forces you to switch by tapping on Scripts for example, just to leave the automation. Luckily the _onbeforeunload_ isn't blocking/preventing that.

**Reply by KTibow** [reply]
What do you mean? Screenshot? Feel free to open a new issue.

**Reply by Recte** [reply]
I will submit an issue, but FYI. When I tap Leave in the image below, I return to the automation and don’t leave. 

This screen shows when I tap the back arrow when in the automation edit screen. 

![FDB12F5F-798A-4923-AEA7-F3A85A09ABC5](https://user-images.githubusercontent.com/26491601/112936060-591fb600-9125-11eb-90ea-a5be606d8424.jpeg)

**Reply by SeanPM5** [reply]
I have an issue open for this already here: https://github.com/home-assistant/frontend/issues/7008
